Unable to create node groups after creating a new cluster (0.14.0)

**What happened?**
eksctl 0.14.0 is unable to create a new node group after successfully creating a cluster with an initial nodegroup.  This error occurs when using `nodeGroup` or `managagedNodeGroup`.  The command fails if there is not a Fargate policy present in the cluster CloudFormation stack output.

```
eksctl create nodegroup -f cluster.yaml
[ℹ]  eksctl version 0.14.0
[ℹ]  using region us-east-1
[ℹ]  will use version 1.14 for new nodegroup(s) based on control plane version
[ℹ]  1 nodegroup(s) that already exist (staging-cluster-workers-v1) will be excluded
[ℹ]  1 nodegroup (staging-cluster-workers-v2) was included (based on the include/exclude rules)
[ℹ]  combined exclude rules: staging-cluster-workers-v1
[ℹ]  no nodegroups present in the current set were excluded by the filter
[ℹ]  will create a CloudFormation stack for each of 1 managed nodegroups in cluster "staging-cluster"
[ℹ]  2 sequential tasks: { fix cluster compatibility, 1 task: { 1 task: { create managed nodegroup "staging-cluster-workers-v2" } } }
[ℹ]  checking cluster stack for missing resources
[ℹ]  cluster stack is missing resources for Fargate
[ℹ]  adding missing resources to cluster stack
[ℹ]  re-building cluster stack "eksctl-staging-cluster-cluster"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x227cc61]

goroutine 87 [running]:
github.com/weaveworks/eksctl/pkg/cfn/builder.(*ClusterResourceSet).addOutputsForVPC(0xc0000cc320)
	github.com/weaveworks/eksctl@/pkg/cfn/builder/vpc.go:166 +0x101
github.com/weaveworks/eksctl/pkg/cfn/builder.(*ClusterResourceSet).AddAllResources(0xc0000cc320, 0xc0004019a0, 0xc0003df6b0)
	github.com/weaveworks/eksctl@/pkg/cfn/builder/cluster.go:68 +0xac
github.com/weaveworks/eksctl/pkg/cfn/manager.(*StackCollection).AppendNewClusterStackResource(0xc0004b8960, 0x100, 0x0, 0x0, 0x0)
	github.com/weaveworks/eksctl@/pkg/cfn/manager/cluster.go:116 +0x362
github.com/weaveworks/eksctl/pkg/cfn/manager.(*StackCollection).FixClusterCompatibility(0xc0004b8960, 0xc0006646e8, 0x100493d)
	github.com/weaveworks/eksctl@/pkg/cfn/manager/compat.go:64 +0x342
github.com/weaveworks/eksctl/pkg/cfn/manager.(*clusterCompatTask).Do(0xc00062c3e0, 0xc0008b4a20, 0x0, 0x0)
	github.com/weaveworks/eksctl@/pkg/cfn/manager/tasks.go:190 +0x6b
github.com/weaveworks/eksctl/pkg/cfn/manager.doSingleTask(0xc0008b49c0, 0x3f0ad20, 0xc00062c3e0, 0x1357ac0)
	github.com/weaveworks/eksctl@/pkg/cfn/manager/tasks.go:265 +0xec
github.com/weaveworks/eksctl/pkg/cfn/manager.doSequentialTasks(0xc0008b49c0, 0xc00062c480, 0x2, 0x2)
	github.com/weaveworks/eksctl@/pkg/cfn/manager/tasks.go:295 +0x70
created by github.com/weaveworks/eksctl/pkg/cfn/manager.(*TaskTree).DoAllSync
	github.com/weaveworks/eksctl@/pkg/cfn/manager/tasks.go:118 +0x237
exit status 2
```

**What you expected to happen?**
After creating a new cluster successfully with an EKS managed node group I should be able to create or delete nodegroups without a Fargate policy or profile.

**How to reproduce it?**
1. Create a new cluster in an existing VPC with one `nodeGroup` or `managedNodeGroup` with the following [config](https://gist.github.com/nathanialg/1d6d5712a8f4a3d6386c04ed3801e82e#file-01-cluster-yaml)
```
eksctl create cluster -f cluster.yaml
```
2. Update the name of the node group in the [config](https://gist.github.com/nathanialg/1d6d5712a8f4a3d6386c04ed3801e82e#file-03-cluster-update-yaml)
3. Create a new nodegroup from the cluster config
```
eksctl create nodegroup -f cluster.yaml
```

**Versions**
Please paste in the output of these commands:
```
$ eksctl version
0.14.0
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.8", GitCommit:"211047e9a1922595eaa3a1127ed365e9299a6c23", GitTreeState:"clean", BuildDate:"2019-10-15T12:11:03Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.9-eks-502bfb", GitCommit:"502bfb383169b124d87848f89e17a04b9fc1f6f0", GitTreeState:"clean", BuildDate:"2020-02-07T01:31:02Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
$ aws --version
aws-cli/2.0.0 Python/3.8.1 Darwin/18.7.0 botocore/2.0.0dev4
```

**Logs**
[create cluster logs](https://gist.github.com/nathanialg/1d6d5712a8f4a3d6386c04ed3801e82e#file-02-cluster-create-log)
[create nodegroup logs](https://gist.github.com/nathanialg/1d6d5712a8f4a3d6386c04ed3801e82e#file-04-nodegroup-create-log)

--- a/pkg/cfn/builder/vpc.go
+++ b/pkg/cfn/builder/vpc.go
@@ -163,7 +163,9 @@ func (c *ClusterResourceSet) addOutputsForVPC() {
 		c.spec.VPC.ID = v
 		return nil
 	})
-	c.rs.defineOutputWithoutCollector(outputs.ClusterFeatureNATMode, c.spec.VPC.NAT.Gateway, false)
+	if c.spec.VPC.NAT != nil {
+		c.rs.defineOutputWithoutCollector(outputs.ClusterFeatureNATMode, c.spec.VPC.NAT.Gateway, false)
+	}
 	if refs, ok := c.subnets[api.SubnetTopologyPrivate]; ok {
 		c.rs.defineJoinedOutput(outputs.ClusterSubnetsPrivate, refs, true, func(v string) error {
 			return vpc.ImportSubnetsFromList(c.provider, c.spec, api.SubnetTopologyPrivate, strings.Split(v, ","))

